pkg/log/json.go

```diff
@@ -0,0 +1,228 @@
+	if log.file != nil {
+		log.file.Infof(format, args...)
+	}
+}
+
+// Error writes a error-level log
+func (w *JSONWriter) Error(args ...interface{}) {
+	log.out.Error(args...)
+	if log.file != nil {
+		log.file.Error(args...)
+	}
+}
+
+// Errorf writes a error-level log with a format
+func (w *JSONWriter) Errorf(format string, args ...interface{}) {
+	log.out.Errorf(format, args...)
+	if log.file != nil {
+		log.file.Errorf(format, args...)
+	}
+}
+
+// Fatalf writes a error-level log with a format
+func (w *JSONWriter) Fatalf(format string, args ...interface{}) {
+	if log.file != nil {
+		log.file.Errorf(format, args...)
+	}
+
+	log.out.Fatalf(format, args...)
+}
+
+// Green writes a line in green
+func (w *JSONWriter) Green(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf(format, args...))
+}
+
+// Yellow writes a line in yellow
+func (w *JSONWriter) Yellow(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf(format, args...))
+}
+
+// Success prints a message with the success symbol first, and the text in green
+func (w *JSONWriter) Success(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf("%s %s", successSymbol, fmt.Sprintf(format, args...)))
+}
+
+// Information prints a message with the information symbol first, and the text in blue
+func (w *JSONWriter) Information(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf("%s %s", informationSymbol, fmt.Sprintf(format, args...)))
+}
+
+// Question prints a message with the question symbol first, and the text in magenta
+func (w *JSONWriter) Question(format string, args ...interface{}) error {
+	return fmt.Errorf("can't ask questions on json mode")
+}
+
+// Warning prints a message with the warning symbol first, and the text in yellow
+func (w *JSONWriter) Warning(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf("%s %s", warningSymbol, fmt.Sprintf(format, args...)))
+}
+
+// Hint prints a message with the text in blue
+func (w *JSONWriter) Hint(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	w.Fprintln(fmt.Sprintf("%s", fmt.Sprintf(format, args...)))
+}
+
+// Fail prints a message with the error symbol first, and the text in red
+func (w *JSONWriter) Fail(format string, args ...interface{}) {
+	log.out.Infof(format, args...)
+	msg := fmt.Sprintf("%s %s", errorSymbol, fmt.Sprintf(format, args...))
+	fmt.Fprintln(w.out.Out, convertToJSON("error", log.stage, msg))
+}
+
+// Println writes a line with colors
+func (w *JSONWriter) Println(args ...interface{}) {
+	w.Fprintln(args...)
+}
+
+// Fprintf prints a line with format
+func (w *JSONWriter) Fprintf(format string, a ...interface{}) {
+	w.Printf(format, a...)
+}
+
+// Fprintln prints a line with format
+func (w *JSONWriter) Fprintln(args ...interface{}) {
+	msg := fmt.Sprint(args...)
```

Comment:
I see `log.Println()` being used in many places just to insert a new line, which is not needed when output format is json. I would add a clause here to return if msg length is zero. I don't find useful a log message as the following:

```json
{"level":"info","stage":"","msg":""}
```

Reply:
Logrus does inject an empty logline, with timestamp and level if no args are provided.

https://go.dev/play/p/Vg-jM2-rw-N

Maybe we should not use at all these `fmt.Println()` statements.